cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -25,9 +25,13 @@ import (
 )
 
 const (
-	machineDeleteAnnotationKey = "cluster.k8s.io/delete-machine"
-	machineAnnotationKey       = "cluster.k8s.io/machine"
-	debugFormat                = "%s (min: %d, max: %d, replicas: %d)"
+	// deprecatedMachineDeleteAnnotationKey should not be removed until minimum cluster-api support is v1alpha3
+	deprecatedMachineDeleteAnnotationKey = "cluster.k8s.io/delete-machine"
+	// TODO: determine what currently relies on deprecatedMachineAnnotationKey to determine when it can be removed
```

Comment:
It wasn't actually clear to me where this is being used. @elmiko do you know? I would expect any v1alpha3 compatible infrastructure providers to be setting ProviderID. We'll still likely need to keep this for backward compatibility, but would like to have a target removal in mind.

Reply:
not sure i completely understand, but afaik, we _should_ be using the const variable names and not the actual string values anywhere. as for how it is used in the context of the autoscaler, the `DeleteNodes` function wraps this all to be used during the scale down attempts from the core.

i was under the impression that these annotations are then used to help inform during the scale down, but perhaps  we don't need them from CAPI side? (if so, it certainly seems like we could investigate removing them)

Reply:
I think where I added the comment with the github UI might be messing up the context I was trying to set a bit. My comment (and the TODO is for the `deprecatedMachineAnnotationKey` rather than `deprecatedMachineDeleteAnnotationKey`.

From what I can tell we are using that annotation for matching a Node to a Machine (it's set on the Node) when a providerID is not present on the Machine to be able to use for matching. It is currently being used by some tests in this repo, but I wasn't sure if there is some known usage externally from pre-v1alpha3 cluster-api infrastructure providers.

Reply:
ok, that makes so much more sense lol =)

i think you are correct, this might be an artifact from previous generations. definitely seems like we are using it for storage of the id. as for the external usage, i'm not 100% sure but i can do a little research. certainly seems like a good candidate to cleanup.